api-controller/graphs.js

```diff
@@ -0,0 +1,80 @@
+'use strict'
+
+const express = require('express');
+const router = express.Router();
+const sampleData = require('../sample-data/rc2.json');
+const multer = require('multer');
+const upload = multer();
+
+function readFile(filename) {
+  return JSON.parse(filename);
+}
+
+// POST /graph/burndown
+// Analyse the received raincatchDump.json & return a json dataset to feed a sprint burndown chart
+router.post('/burndown', function (req, res) {
+  //res.send('You have just hit the route: /graph/burndown');
+  let data = require('../sample-data/rc2.json');
+  let sprintTickets = data.map(function (obj) {
+
+  });
+
+  res.json(sampleData);
+  // TODO overview - accept & validate name of sprint & return json dataset to populate a burndown chart returned
+  // TODO retrieve date range for sprint
+  // TODO retrieve all issues that correspond to dates in sprint
+  // TODO map all issues by date and storypoint value
+  // TODO take account of any issues that drop in & out of sprint
+  // TODO validate name of sprint
+});
+
+// POST /graph/sprintReport
+// Analyse the received raincatchDump.json & return a json dataset to feed a sprint report
+router.post('/sprintReport', function (req, res) {
+  res.send('You have just hit the route: /graph/sprintReport');
+  // TODO overview - accept & validate name of a sprint & return json dataset to populate a JIRA sprint reports
+  // TODO return same data as for /burndown along with data table of issues in burndown
+});
+
+// POST /graph/velocity
```

Comment:
Yeah, definitely, lot of stubs in here that the implementations will be covered in additional tickets